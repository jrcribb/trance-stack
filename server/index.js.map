{
  "version": 3,
  "sources": ["../server.js", "server-entry-module:@remix-run/dev/server-build", "../app/entry.server.tsx", "../app/root.tsx", "../app/routes/index.tsx", "server-assets-manifest:@remix-run/dev/assets-manifest"],
  "sourcesContent": ["import { createRequestHandler } from \"@remix-run/architect\";\nimport * as build from \"@remix-run/dev/server-build\";\n\nexport const handler = createRequestHandler({\n  build,\n  mode: process.env.NODE_ENV,\n});\n", "\nimport * as entryServer from \"./entry.server.tsx\";\nimport * as route0 from \"./root.tsx\";\nimport * as route1 from \"./routes/index.tsx\";\n  export { default as assets } from \"@remix-run/dev/assets-manifest\";\n  export const assetsBuildDirectory = \"public/build\";\n  export const future = {\"unstable_cssModules\":false,\"unstable_cssSideEffectImports\":false,\"unstable_dev\":false,\"unstable_vanillaExtract\":false,\"v2_errorBoundary\":false,\"v2_meta\":false,\"v2_routeConvention\":false};\n  export const publicPath = \"/_static/build/\";\n  export const entry = { module: entryServer };\n  \n  export const routes = {\n    \"root\": {\n      id: \"root\",\n      parentId: undefined,\n      path: \"\",\n      index: undefined,\n      caseSensitive: undefined,\n      module: route0\n    },\n  \"routes/index\": {\n      id: \"routes/index\",\n      parentId: \"root\",\n      path: undefined,\n      index: true,\n      caseSensitive: undefined,\n      module: route1\n    }\n  };", "import type { EntryContext } from '@remix-run/node';\nimport { RemixServer } from '@remix-run/react';\nimport { renderToString } from 'react-dom/server';\n\nexport default (\n  request: Request,\n  responseStatusCode: number,\n  responseHeaders: Headers,\n  remixContext: EntryContext\n) => {\n  const markup = renderToString(\n    <RemixServer context={remixContext} url={request.url} />\n  );\n\n  responseHeaders.set('Content-Type', 'text/html');\n\n  return new Response('<!DOCTYPE html>' + markup, {\n    headers: responseHeaders,\n    status: responseStatusCode\n  });\n};\n", "import type { MetaFunction } from '@remix-run/node';\nimport {\n  Links,\n  LiveReload,\n  Meta,\n  Outlet,\n  Scripts,\n  ScrollRestoration\n} from '@remix-run/react';\n\nexport const meta: MetaFunction = () => ({\n  charset: 'utf-8',\n  title: 'REPL_APP_NAME',\n  viewport: 'width=device-width,initial-scale=1'\n});\n\nexport default function App() {\n  return (\n    <html lang=\"en\">\n      <head>\n        <Meta />\n        <Links />\n      </head>\n      <body>\n        <Outlet />\n        <ScrollRestoration />\n        <Scripts />\n        <LiveReload />\n      </body>\n    </html>\n  );\n}\n", "export default () => (\n  <div>\n    Hello World!\n  </div>\n);\n", "export default {'version':'0ffe8003','entry':{'module':'/_static/build/entry.client-6BKN5BHR.js','imports':['/_static/build/_shared/chunk-IKHP34ID.js','/_static/build/_shared/chunk-TFGU5KND.js']},'routes':{'root':{'id':'root','parentId':undefined,'path':'','index':undefined,'caseSensitive':undefined,'module':'/_static/build/root-3QDZ7QDP.js','imports':undefined,'hasAction':false,'hasLoader':false,'hasCatchBoundary':false,'hasErrorBoundary':false},'routes/index':{'id':'routes/index','parentId':'root','path':undefined,'index':true,'caseSensitive':undefined,'module':'/_static/build/routes/index-2DEIG5UD.js','imports':undefined,'hasAction':false,'hasLoader':false,'hasCatchBoundary':false,'hasErrorBoundary':false}},'cssBundleHref':undefined,'url':'/_static/build/manifest-0FFE8003.js'};"],
  "mappings": ";;;;;;;;;;;;;;;;;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA,uBAAqC;;;ACArC;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;;;ACAA;AAAA;AAAA;AAAA;AACA,mBAA4B,6BAC5B,gBAA+B,6BAS3B,2DAPG,uBAAQ,CACb,SACA,oBACA,iBACA,iBACG;AACH,MAAM,aAAS;AAAA,IACb,mDAAC,4BAAY,SAAS,cAAc,KAAK,QAAQ,OAAjD;AAAA;AAAA;AAAA;AAAA,WAAsD;AAAA,EACxD;AAEA,yBAAgB,IAAI,gBAAgB,WAAW,GAExC,IAAI,SAAS,oBAAoB,QAAQ;AAAA,IAC9C,SAAS;AAAA,IACT,QAAQ;AAAA,EACV,CAAC;AACH;;;ACpBA;AAAA;AAAA;AAAA;AAAA;AACA,IAAAA,gBAOO,6BAYCC,0BAAA,kCAVK,OAAqB,OAAO;AAAA,EACvC,SAAS;AAAA,EACT,OAAO;AAAA,EACP,UAAU;AACZ;AAEe,SAAR,MAAuB;AAC5B,SACE,oDAAC,UAAK,MAAK,MACT;AAAA,wDAAC,UACC;AAAA,0DAAC,wBAAD;AAAA;AAAA;AAAA;AAAA,aAAM;AAAA,MACN,oDAAC,yBAAD;AAAA;AAAA;AAAA;AAAA,aAAO;AAAA,SAFT;AAAA;AAAA;AAAA;AAAA,WAGA;AAAA,IACA,oDAAC,UACC;AAAA,0DAAC,0BAAD;AAAA;AAAA;AAAA;AAAA,aAAQ;AAAA,MACR,oDAAC,qCAAD;AAAA;AAAA;AAAA;AAAA,aAAmB;AAAA,MACnB,oDAAC,2BAAD;AAAA;AAAA;AAAA;AAAA,aAAS;AAAA,MACT,oDAAC,8BAAD;AAAA;AAAA;AAAA;AAAA,aAAY;AAAA,SAJd;AAAA;AAAA;AAAA;AAAA,WAKA;AAAA,OAVF;AAAA;AAAA;AAAA;AAAA,SAWA;AAEJ;;;AC/BA;AAAA;AAAA;AAAA;AACE,IAAAC,0BAAA,kCADK,iBAAQ,MACb,oDAAC,SAAI,4BAAL;AAAA;AAAA;AAAA;AAAA,OAEA;;;ACHF,IAAO,0BAAQ,EAAC,SAAU,YAAW,OAAQ,EAAC,QAAS,2CAA0C,SAAU,CAAC,4CAA2C,0CAA0C,EAAC,GAAE,QAAS,EAAC,MAAO,EAAC,IAAK,QAAO,UAAW,QAAU,MAAO,IAAG,OAAQ,QAAU,eAAgB,QAAU,QAAS,mCAAkC,SAAU,QAAU,WAAY,IAAM,WAAY,IAAM,kBAAmB,IAAM,kBAAmB,GAAK,GAAE,gBAAe,EAAC,IAAK,gBAAe,UAAW,QAAO,MAAO,QAAU,OAAQ,IAAK,eAAgB,QAAU,QAAS,2CAA0C,SAAU,QAAU,WAAY,IAAM,WAAY,IAAM,kBAAmB,IAAM,kBAAmB,GAAK,EAAC,GAAE,eAAgB,QAAU,KAAM,sCAAqC;;;AJK5wB,IAAM,uBAAuB,gBACvB,SAAS,EAAC,qBAAsB,IAAM,+BAAgC,IAAM,cAAe,IAAM,yBAA0B,IAAM,kBAAmB,IAAM,SAAU,IAAM,oBAAqB,GAAK,GACpM,aAAa,mBACb,QAAQ,EAAE,QAAQ,qBAAY,GAE9B,SAAS;AAAA,EACpB,MAAQ;AAAA,IACN,IAAI;AAAA,IACJ,UAAU;AAAA,IACV,MAAM;AAAA,IACN,OAAO;AAAA,IACP,eAAe;AAAA,IACf,QAAQ;AAAA,EACV;AAAA,EACF,gBAAgB;AAAA,IACZ,IAAI;AAAA,IACJ,UAAU;AAAA,IACV,MAAM;AAAA,IACN,OAAO;AAAA,IACP,eAAe;AAAA,IACf,QAAQ;AAAA,EACV;AACF;;;ADxBK,IAAM,cAAU,uCAAqB;AAAA,EAC1C;AAAA,EACA,MAAM;AACR,CAAC;",
  "names": ["import_react", "import_jsx_dev_runtime", "import_jsx_dev_runtime"]
}
