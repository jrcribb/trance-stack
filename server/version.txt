f4bd41e6