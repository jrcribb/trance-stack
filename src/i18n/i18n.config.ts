import type { InitOptions } from 'i18next';

export const supportedLngs = ['en'];
export const defaultNS = ['translation'];
export const fallbackLng = 'en';

export const getBaseClientConfig = (): InitOptions => ({
  supportedLngs: ['en'],
  fallbackLng: fallbackLng,
  defaultNS: defaultNS,
  interpolation: { escapeValue: false },
  react: { useSuspense: false },
  detection: {
    order: ['htmlTag'],
    caches: []
  }
});
