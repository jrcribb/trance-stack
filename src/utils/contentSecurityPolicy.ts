import { getCSP, SELF, UNSAFE_INLINE, BLOB, DATA, NONE } from 'csp-header';

/**
 * @name contentSecurityPolicy
 * @description A better way to set CSP directives
 */
export const contentSecurityPolicy = (isDevelopment: boolean): string => {
  return getCSP({
    directives: {
      'default-src': [SELF],
      'worker-src': [BLOB],
      'script-src': [
        SELF,
        UNSAFE_INLINE,
        'https://*.cookieyes.com',
        'https://cdn-cookieyes.com',
        'https://*.mixpanel.com',
        'https://*.hotjar.com',
        'https://*.googletagmanager.com',
        'https://*.google-analytics.com',
        'https://*.hotjar.io'
      ],
      'connect-src': [
        SELF,
        UNSAFE_INLINE,
        'https://*.mixpanel.com',
        'https://*.cookieyes.com',
        'https://cdn-cookieyes.com',
        'https://*.googletagmanager.com',
        'https://*.google-analytics.com',
        'wss://*.hotjar.com',
        'https://*.hotjar.com',
        'https://*.hotjar.io',
        isDevelopment ? 'ws://localhost:2222' : '',
        isDevelopment ? 'ws://localhost:8002' : ''
      ],
      'img-src': ['https:', DATA, 'http:'],
      'frame-ancestors': [NONE],
      'frame-src': [
        'https://*.mixpanel.com',
        'https://*.cookieyes.com',
        'https://cdn-cookieyes.com',
        'https://*.googletagmanager.com',
        'https://*.google-analytics.com',
        'https://*.hotjar.com',
        'https://*.hotjar.io'
      ],
      'style-src': [
        UNSAFE_INLINE,
        SELF
      ]
    }
  });
};
