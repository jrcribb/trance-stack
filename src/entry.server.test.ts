import crypto from 'node:crypto';
import { Response } from '@remix-run/node';
import { renderToString } from 'react-dom/server';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import { initServerI18n } from '~/i18n';
import { addSecurityHeaders, sanitizeHeaders } from '~/utils/securityHeaders';
import entry, { handleDataRequest } from './entry.server';

vi.mock('~/i18n');
vi.mock('~/session.server');
vi.mock('~/utils/securityHeaders');
vi.mock('@remix-run/react');
vi.mock('react-dom/server');
vi.mock('node:crypto');

describe('entry.server', () => {
  beforeEach(() => {
    vi.resetAllMocks();
    vi.mocked(initServerI18n).mockResolvedValue('mocked initServerI18n' as never);
    vi.mocked(renderToString).mockReturnValue('mocked renderToString markup');
    vi.mocked(crypto.randomBytes).mockReturnValue('mocked nonce' as never);
  });

  it('should return the markup', async () => {
    const request = new Request('https://example.com');
    const responseCode = 201;
    const responseHeaders = new Headers();
    const context = {
      staticHandlerContext: {
        loaderData: {
          root: {
            locale: 'en'
          }
        }
      }
    } as never;

    const actualResponse = await entry(request, responseCode, responseHeaders, context);

    expect(await actualResponse.text()).toMatchInlineSnapshot('"<!DOCTYPE html>mocked renderToString markup"');
    expect(actualResponse.status).toBe(201);
  });

  it('should initialise things correctly', async () => {
    vi.stubEnv('NODE_ENV', 'development');

    const request = new Request('https://example.com');
    const responseCode = 200;
    const responseHeaders = new Headers();
    const context = {
      staticHandlerContext: {
        loaderData: {
          root: {
            locale: 'en'
          }
        }
      }
    } as never;

    await entry(request, responseCode, responseHeaders, context);

    expect(initServerI18n).toHaveBeenCalledWith('en', context);
    expect(addSecurityHeaders).toHaveBeenCalledWith(responseHeaders, 'mocked nonce');
    expect(sanitizeHeaders).toHaveBeenCalledWith(responseHeaders);

  });

  it('should return the correct headers', async () => {
    vi.stubEnv('NODE_ENV', 'production');

    const request = new Request('https://example.com');
    const responseCode = 200;
    const responseHeaders = new Headers();
    const context = {
      staticHandlerContext: {
        loaderData: {
          root: {
            locale: 'en'
          }
        }
      }
    } as never;

    const actualResponse = await entry(request, responseCode, responseHeaders, context);
    expect(addSecurityHeaders).toHaveBeenCalledWith(responseHeaders, 'mocked nonce');
    expect(actualResponse.status).toBe(200);
    expect(await actualResponse.headers).toMatchInlineSnapshot(`
      Headers {
        Symbol(query): [
          "cache-control",
          "no-cache, max-age=0, s-maxage=0",
          "content-type",
          "text/html",
        ],
        Symbol(context): null,
      }
    `);

  });

  describe('when the request is for data loading', () => {
    describe('and the upstream loader hasn\'t overriden it', () => {
      it('should set the cache control to 10 minutes', () => {
        const request = new Request('https://example.com/_remix/data');
        const upstreamResponse = new Response(null, {
          headers: {}
        });
        const actual = handleDataRequest(upstreamResponse, { request: request });

        expect(actual.headers.has('Cache-Control')).toBeTruthy();
        expect(actual.headers.get('Cache-Control')).toEqual('private, max-age=600, no-cache="Set-Cookie"');
      });
    });

    describe('and the upstream loader has overriden it', () => {
      it('should set the cache control to 10 minutes', () => {
        const request = new Request('https://example.com/_remix/data');
        const upstreamResponse = new Response(null, {
          headers: {
            'Cache-Control': 'set-in-upstream-loader'
          }
        });
        const actual = handleDataRequest(upstreamResponse, { request: request });

        expect(actual.headers.has('Cache-Control')).toBeTruthy();
        expect(actual.headers.get('Cache-Control')).toEqual('set-in-upstream-loader');
      });
    });
  });
});
