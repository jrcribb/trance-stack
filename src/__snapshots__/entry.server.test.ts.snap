// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`entry.server > should return the correct headers 1`] = `
Headers {
  Symbol(query): [
    "content-type",
    "text/html",
    "cache-control",
    "no-cache, max-age=0, s-maxage=0",
  ],
  Symbol(context): null,
}
`;
